### PR TITLE
SAK-50284 Kernel email Message-ID must contain angle brackets

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java [hunk 1]
```diff
@@ -1608,9 +1608,9 @@ else if (canUseCharset(message, CharacterSet.WINDOWS_1252) && canUseCharset(getS
 		/**
 		 * Override MimeMessage Message-ID to use Sakai serverId instead of hostname
 		 * https://javaee.github.io/javamail/FAQ#msgid
-		 * @throws MessagingException
-		 */
-		protected void updateMessageID() throws MessagingException
+         */
+		@Override
+        protected void updateMessageID() throws MessagingException
```
Comment (jesusmmp) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java:1613]: Spacing

File: kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java [hunk 2]
```diff
@@ -1621,10 +1621,11 @@ protected void updateMessageID() throws MessagingException
 					append(System.currentTimeMillis()).
 					append('@').
 					append(serverConfigurationService.getServerName());
-			setHeader("Message-ID", s.toString());
+			setHeader("Message-ID", "<" + s + ">");
 		}
 
-		protected void updateHeaders() throws MessagingException
+		@Override
+        protected void updateHeaders() throws MessagingException
```
Comment (jesusmmp) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java:1628]: Spacing

File: kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java [hunk 3]
```diff
@@ -1672,17 +1673,12 @@ private void addEncodedHeader(String header, String name) throws MessagingExcept
...
-			 catch (MessagingException e) 
-			 {
-				  log.error("Email.MyMessage: exception: " + e, e);
-				  addHeaderLine(header);
-			 } 
-			 catch (UnsupportedEncodingException e)
+			 catch (MessagingException | UnsupportedEncodingException e)
 			 {
-				  log.error("Email.MyMessage: exception: " + e, e);
+                 log.error("Email.MyMessage: exception: {}", e, e);
```
Comment (jesusmmp) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java:1678]: Spacing

File: kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java [hunk 4]
```diff
@@ -1692,7 +1688,6 @@ public String propName(String propNameTemplate)
 
 	public String propName(String propNameTemplate, String protocol)
 	{
-		String formattedName = String.format(propNameTemplate, protocol);
-		return formattedName;
+        return String.format(propNameTemplate, protocol);
```
Comment (jesusmmp) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java:1691]: Spacing

File: kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java [hunk 5]
```diff
@@ -1343,7 +1343,7 @@ protected void sendMessageAndLog(InternetAddress[] to, long start, MimeMessage m
 					buf.append(" with ").append(parts.getCount() - 1).append(" attachments");
 				}
 			}
-			catch (IOException ioe) {}
+			catch (IOException ignored) {}
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java:1346]:
```suggestion
			catch (IOException ioe) {
			    log.debug("could not read message contents, {}", ioe.toString());
			}
```

**Review by ern — APPROVED** [review]

File: kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java [hunk 6]
```diff
@@ -1608,9 +1610,9 @@ else if (canUseCharset(message, CharacterSet.WINDOWS_1252) && canUseCharset(getS
 		/**
 		 * Override MimeMessage Message-ID to use Sakai serverId instead of hostname
 		 * https://javaee.github.io/javamail/FAQ#msgid
-		 * @throws MessagingException
 		 */
-		protected void updateMessageID() throws MessagingException
+		@Override
+		 protected void updateMessageID() throws MessagingException
```
Comment (adrianfish) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java:1615]:
```suggestion
		protected void updateMessageID() throws MessagingException
```

File: kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java [hunk 7]
```diff
@@ -1621,10 +1623,11 @@ protected void updateMessageID() throws MessagingException
 					append(System.currentTimeMillis()).
 					append('@').
 					append(serverConfigurationService.getServerName());
-			setHeader("Message-ID", s.toString());
+			setHeader("Message-ID", "<" + s + ">");
 		}
 
-		protected void updateHeaders() throws MessagingException
+		@Override
+		 protected void updateHeaders() throws MessagingException
```
Comment (adrianfish) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/email/impl/BasicEmailService.java:1630]:
```suggestion
		protected void updateHeaders() throws MessagingException
```

**Pull request merged**